CMakeLists.txt

```diff
@@ -131,6 +131,21 @@ add_cxx_compiler_option ("-Wextra")
 add_cxx_compiler_option ("-Wno-overloaded-virtual")
 add_cxx_compiler_option ("-Wno-deprecated")
 
+# If we're on GCC, use the gold linker if available.
+if(CMAKE_COMPILER_IS_GNUCC)
+    execute_process(
+            COMMAND ${CMAKE_C_COMPILER} -fuse-ld=gold -Wl,--version
+            ERROR_QUIET OUTPUT_VARIABLE LD_VERSION)
+    if ("${LD_VERSION}" MATCHES "GNU gold")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fuse-ld=gold")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fuse-ld=gold")
+        message("-- Using the GNU gold linker.")
```

Comment:
message (STATUS "Using the GNU gold linker.") will produce the "--" prefixed string, properly indented.

Reply:
I'll make that change.